smoke/regexp/b.rb

```diff
@@ -10,7 +10,7 @@
   # @type var match_ref_5: Array<String>
   match_ref_5 = match[0..1]
 
-  # @type var begin_1: Integer
+  # @type var begin_1: Integer?
```

Comment:
To test the type of expression, the following would be correct and easy.

1. Assign to a variable without annotation
2. Call undefined method
3. Test the error message

Reply:
Example: https://github.com/soutaro/steep/blob/3a12024a089dab359ae05580adf2670dc3b5008c/smoke/type_case/a.rb#L13

Reply:
As you pointed, I tested only normal cases, not abnormal cases. 😓 

When we test type definitions, the following points are necessary, right?

1. Valid argument(s) and return type of a method. For example:
    ```
    # @type var begin_1: Integer?
    begin_1 = match.begin(0)
    ```
2. Invalid argument(s) type of a method. For example:
    ```
    # !expects ArgumentTypeMismatch: receiver=::MatchData, expected=::Integer, actual=nil
    match.begin(nil)
    ```